rest_server/RestServer.go

```diff
@@ -0,0 +1,269 @@
+	var config Config
+
+	// Parse Siodb instance parameters file
+	if config, err = loadInitFile(SiodbInstanceConfigurationPath + "/config"); err != nil {
+		log.Fatalln(err)
+	}
+
+	// Validate and load parameters
+	if err = restServer.loadConfig(SiodbInstanceConfigurationPath, config); err != nil {
+		log.Fatalln(err)
+	}
+
+	// Logging setup
+	siodbLoggerPool.AddSiodbLogger(restServer.logConsoleSeverity, restServer.logConsoleDestination)
+	siodbLoggerPool.AddSiodbLogger(restServer.logFileSeverity, restServer.logFileDestination)
+
+	siodbLoggerPool.Output(INFO, "Starting with config: %v", restServer)
+
+	// Gin Log at global level
+	siodbLoggerPool.ConfigGinLogger()
+
+	// Connection pool to IOMgr
+	IOMgrCPool, err = CreateIOMgrConnPool(4, 8, config)
+	if err != nil {
+		siodbLoggerPool.Output(FATAL, "Cannot create connection pool: %v", err)
+	}
+
+	// Start routers
+	if restServer.Ipv4HTTPPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.router.New(restServer.Ipv4HTTPPort); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create router: %v", err)
+			}
+			if err := restWorker.CreateRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServer.Ipv4HTTPSPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.router.New(restServer.Ipv4HTTPSPort); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create router: %v", err)
+			}
+			if err := restWorker.CreateRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPSRouter(restServer.TLSCertificate, restServer.TLSPrivateKey); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServer.Ipv6HTTPPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.router.New(restServer.Ipv6HTTPPort); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create router: %v", err)
+			}
+			if err := restWorker.CreateRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot start router: %v", err)
+			}
+		}()
+	}
+	if restServer.Ipv6HTTPSPort != 0 {
+		go func() {
+			var restWorker RestWorker
+			if err := restWorker.router.New(restServer.Ipv6HTTPSPort); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create router: %v", err)
+			}
+			if err := restWorker.CreateRouter(); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot create route: %v", err)
+			}
+			if err := restWorker.StartHTTPSRouter(restServer.TLSCertificate, restServer.TLSPrivateKey); err != nil {
+				siodbLoggerPool.Output(FATAL, "Cannot start router: %v", err)
+			}
+		}()
+	}
+
+	<-make(chan int)
```

Comment:
what's this?

Reply:
This is to block indefinitely. I changed that for the next push to include signals handling.